HERACLES-17 / HERACLES-19: Holter form

marta- opened the pull request:
Form specs: https://phenotips.atlassian.net/browse/HERACLES-18

marta- committed:
HERACLES-17 / HERACLES-19: Holter form

sashaandjic commented:
Not happy with `Symptom duration` implementation.
It should be a time interval rather than time (as per requirement), or number of hours and minutes.
![image](https://github.com/data-team-uhn/cards/assets/23243244/bbb4feec-8c2d-4ea1-87f3-bd1d395a0a35)

sashaandjic commented:
New XML provided with ` symptom_duration` is now `text `rather than `time `,with regexp `^\d{2}:[0-5]\d$` to allow duration of more than a day

marta- committed:
HERACLES-17 / HERACLES-19: Holter form
Updated the data type for the symptom_duration question

marta- commented on heracles-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Holter.xml:
```diff
@@ -0,0 +1,2581 @@
+			<name>description</name>
+			<value>Triggered</value>
+			<type>String</type>
+		</property>
```
```suggestion
```

marta- committed:
HERACLES-17 / HERACLES-19: Holter form

Remove description from `holter_yearnumber` question.

Hajany commented:
_Reason_ should show up if _Partially_ or _Did not complete_ are selected above, right now it only shows up if _Did not complete_ is selected
![image](https://github.com/data-team-uhn/cards/assets/96542282/f686a72f-fb1c-4b0d-a7c5-26809bf333c0)

marta- committed:
HERACLES-17 / HERACLES-19: Holter form
* Fix conditions depending on completion status
* Use textbox instead of input + notes for comments

sashaandjic reviewed (approved):
tested

marta- commented:
Rebased on the latest dev, where the TimeQuestion fields are as wide as all others.

sashaandjic reviewed (approved):
re-tested

marta- committed:
HERACLES-17 / HERACLES-19: Holter form
Implementing post-demo feedback: updated question texts, reordering, updated conditions

marta- committed:
HERACLES-17 / HERACLES-19: Holter form
Removed unnecessary wrapper section

sashaandjic reviewed (approved):
tested

Hajany commented:
Additional physician consult comments should be optional and Additional physician consult abnormal comments should be mandatory
![image](https://github.com/data-team-uhn/cards/assets/96542282/d2da09ba-bb12-4e3c-b1a8-5d5160c2285f)

marta- commented on heracles-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Holter.xml:
```diff
@@ -0,0 +1,2557 @@
+			</property>
+			<property>
+				<name>minAnswers</name>
+				<value>0</value>
```
```suggestion
				<value>1</value>
```

marta- commented on heracles-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Holter.xml:
```diff
@@ -0,0 +1,2557 @@
+		</property>
+		<property>
+			<name>minAnswers</name>
+			<value>1</value>
```
```suggestion
			<value>0</value>
```

marta- committed:
HERACLES-17 / HERACLES-19: Holter form

Addressing review: Fixed the minAnswers of the two "additional comments" questions

marta- commented:
> Additional physician consult comments should be optional and Additional physician consult abnormal comments should be mandatory 

@Hajany fixed, please confirm.

marta- commented:
> Additional physician consult comments should be optional and Additional physician consult abnormal comments should be mandatory

@Hajany Fixed, please confirm.

sashaandjic reviewed (approved):
tested
